lib/env/environment.rb

```diff
@@ -65,21 +65,44 @@ def [](k)
       end
 
       def default
-        Config.user_data.fetch(:default_environment)
+        Config.user_data.fetch(:default_environment) || begin
+          unless Config.user_data.fetch(:system_default_opt_out)
+            Config.data.fetch(:default_environment)
+          end
+        end
       end
 
-      def remove_default
-        Config.user_data.delete(:default_environment)
-        Config.save_user_data
+      def remove_default(global)
+        if global
+          Config.data.delete(:default_environment)
+          Config.save_data
+        else
+          Config.user_data.delete(:default_environment)
+          Config.save_user_data
+        end
       end
 
-      def set_default(env_name)
+      def set_default(env_name, global)
```

Comment:
There are some `global`s here that should be `system` please.  Including the error message.